samples_project/Assets/SampleViewer/Samples/FeatureLayer/FeatureLayer.cs

```diff
@@ -0,0 +1,407 @@
+                && UIManager.DisplayText != FeatureLayerUIManager.TextToDisplay.IndexOutOfBoundsError)
+            {
+                UIManager.DisplayText = FeatureLayerUIManager.TextToDisplay.Information;
+            }
+        }
+        else
+        {
+            UIManager.DisplayText = FeatureLayerUIManager.TextToDisplay.CoordinatesError;
+        }
+    }
+
+    private void EmptyOutfieldsDropdown()
+    {
+        if (ListItems != null)
+        {
+            outfields.Clear();
+            foreach (var item in ListItems)
+            {
+                Destroy(item.gameObject);
+            }
+
+            ListItems.Clear();
+        }
+    }
+
+    private void PopulateOutfieldsDropdown(string Response)
+    {
+        var jObject = JObject.Parse(Response);
+        var jFeatures = jObject.SelectToken("features").ToArray();
+        var properties = jFeatures[0].SelectToken("properties");
+        //Populate Outfields drop down
+        foreach (var outfield in properties)
+        {
+            if (outfields.Contains("Get All Features"))
+            {
+                var key = outfield.ToString();
+                var props = key.Split(":");
+                outfields.Add(props[0]);
+                var item = Instantiate(OutfieldItem);
+                item.tag = "ToggleItem";
+                ListItems.Add(item.GetComponent<Toggle>());
+                item.GetComponentInChildren<TextMeshProUGUI>().text = props[0];
+                item.transform.SetParent(contentContainer);
+                item.transform.localScale = Vector2.one;
+            }
+            else
+            {
+                outfields.Add("Get All Features");
+                var item = Instantiate(OutfieldItem);
+                item.tag = "ToggleItem";
+                ListItems.Add(item.GetComponent<Toggle>());
+                item.GetComponentInChildren<TextMeshProUGUI>().text = "Get All Features";
+                item.transform.SetParent(contentContainer);
+                item.transform.localScale = Vector2.one;
+            }
+        }
+    }
+
+    private void MoveCamera()
+    {
+        if (FeatureItems.Count != 0)
+        {
+            if (GetAllFeatures)
+            {
+                var cameraLocationComponent = arcGISCamera.gameObject.GetComponent<ArcGISLocationComponent>();
+                var position = new ArcGISPoint(FeatureItems[0].GetComponent<ArcGISLocationComponent>().Position.X,
+                    FeatureItems[0].GetComponent<ArcGISLocationComponent>().Position.Y, 10000, cameraLocationComponent.Position.SpatialReference);
+                cameraLocationComponent.Position = position;
+                cameraLocationComponent.Rotation = new ArcGISRotation(cameraLocationComponent.Rotation.Heading, 0.0,
+                    cameraLocationComponent.Rotation.Roll);
+            }
+            else
+            {
+                var cameraLocationComponent = arcGISCamera.gameObject.GetComponent<ArcGISLocationComponent>();
```

Comment:
Basically any place you're using `GetComponent()`, try to cache them in awake or start